buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -350,8 +333,13 @@ private void updateComponentOffsets(int cIndex) {
      */
     public CompositeByteBuf removeComponent(int cIndex) {
         checkComponentIndex(cIndex);
-        components.remove(cIndex).freeIfNecessary();
-        updateComponentOffsets(cIndex);
+        Component comp = components.remove(cIndex);
+        int len = comp.length;
```

Comment:
maybe... I just wanted to ensure we not break something once freeIfNecessary may reset all things in Component later.